tests/ssg_test_suite/rule.py

```diff
@@ -68,24 +66,24 @@ def _send_scripts(domain_ip):
     logging.debug("Uploading scripts.")
     log_file_name = os.path.join(LogHelper.LOG_DIR, "data.upload.log")
 
-    command = "ssh {0} mkdir -p {1}".format(machine, remote_dir)
+    command = ("ssh", machine, "mkdir", "-p", remote_dir)
     with open(log_file_name, 'a') as log_file:
         try:
-            subprocess.check_call(shlex.split(command),
+            subprocess.check_call(command,
                                   stdout=log_file,
                                   stderr=subprocess.STDOUT)
         except subprocess.CalledProcessError as e:
             logging.error("Cannot create directory {0}.".format(remote_dir))
             return False
 
-        command = "scp {0} {1}:{2}".format(archive_file, machine, remote_dir)
-        subprocess.check_call(shlex.split(command),
+        command = ("scp", archive_file, "{0}:{1}".format(machine, remote_dir))
+        subprocess.check_call(command,
```

Comment:
Could this be inside a `try .., except`?

Reply:
Sure, and as those calls are very similar, I have made a small function that ensures that they are performed in the same way.